scvi/models/autozivae.py

```diff
@@ -0,0 +1,310 @@
+import torch
+import torch.nn.functional as F
+from torch.distributions import Normal, Beta, Gamma, kl_divergence as kl
+
+
+from scvi.models.log_likelihood import log_zinb_positive, log_nb_positive
+from scvi.models.vae import VAE
+from scipy.special import logit
+from scvi.models.utils import one_hot
+
+torch.backends.cudnn.benchmark = True
+
+
+class AutoZIVAE(VAE):
+    def __init__(
+        self,
+        n_input: int,
+        alpha_prior: float = 0.5,
+        beta_prior: float = 0.5,
+        minimal_dropout: float = 0.01,
+        **args,
+    ):
+
+        if "reconstruction_loss" in args:
+            raise ValueError(
+                "No reconstruction loss must be specified for AutoZI : it is 'autozinb'."
+            )
+
+        super().__init__(n_input, **args)
+        self.reconstruction_loss = "autozinb"
+        self.minimal_dropout = minimal_dropout
+
+        # Parameters of prior Bernoulli Beta distribution : alpha + beta = 1 if only one is specified
+        if beta_prior is None and alpha_prior is not None:
+            beta_prior = 1.0 - alpha_prior
+        if alpha_prior is None and beta_prior is not None:
+            alpha_prior = 1.0 - beta_prior
+
+        device = (
+            torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+        )
+
+        # Create parameters for Bernoulli Beta prior and posterior distributions
+        # Each paramer, whose values are in (0,1), is encoded as its logit, in the set of real numbers
+
+        if self.dispersion == "gene":
```

Comment:
Why is this using the argument dispersion ? This should be a separate parameter called ZI_design or something similar ?

Reply:
Indeed, this was a bad confusion. I just chose `zero_inflation` to be consistent with `dispersion`.